packages/plugin-docusaurus/src/client/theme/SearchBar/index.tsx

```diff
@@ -118,7 +150,42 @@ export default function SearchBar(): JSX.Element {
     return () => {
       search.destroy()
     }
-  }, [isBrowser, siteConfig, indexData])
+  }, [isBrowser, siteConfig, database, colorMode])
+
+  useEffect(() => {
+    async function loadDatabase(): Promise<void> {
+      let data: RawDataWithPositions
+      if (!searchData) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const searchResponse = await fetch(searchBaseUrl.replace('@VERSION@', version!))
+
+        if (searchResponse.status === 0) {
+          throw new Error(`Network error: ${await searchResponse.text()}`)
+        } else if (searchResponse.status !== 200) {
+          throw new Error(`HTTP error ${searchResponse.status}: ${await searchResponse.text()}`)
+        }
+
+        const deflated = ungzip(await searchResponse.arrayBuffer(), { to: 'string' })
```

Comment:
why we should to this by ourselves? if the content encoding is right, the fetch should to this automatically.

Reply:
I had exactly your doubt.
I chose to manually compress to make sure this is always handled correctly so that users get small JSON to be loaded. For instance, in the fastify website the uncompressed data is 8MB while the compressed one is around 1MB.